src/jit/codegencommon.cpp

```diff
@@ -10302,6 +10302,22 @@ void CodeGen::genCaptureFuncletPrologEpilogInfo()
 /*****************************************************************************
  *
  *  Generates code for an EH funclet prolog.
+ *
+ *
+ *  Funclets have the following incoming arguments:
+ *
+ *      catch/filter-handler: eax = the exception object that was caught (see GT_CATCH_ARG), ecx = FP
```

Comment:
Last time I did so and I had to also put `mov eax, edx` in funclet prolog because of 
```
#define REG_EXCEPTION_OBJECT     REG_EAX
```
 in jit/target.h.
So I thought if we pass the Exception Object directly to EAX then we could save an instruction.

I also thought it would be better if we could set REG_EXCEPTION_OBJECT to REG_EDX.